python/pfs/drp/stella/reduceExposure.py

```diff
@@ -230,6 +231,28 @@ def run(self, sensorRefList):
         self.write(sensorRefList, results)
         return results
 
+    def runIsr(self, sensorRef):
+        """Run Instrument Signature Removal (ISR)
+
+        This method exists to allow us to post-process the ISR-ed image.
+
+        Parameters
+        ----------
+        sensorRef : `lsst.daf.persistence.ButlerDataRef`
+            Data reference for sensor.
+
+        Returns
+        -------
+        exposure : `lsst.afw.image.Exposure`
+            Sensor image after ISR has been applied.
+        """
+        exposure = self.isr.runDataRef(sensorRef).exposure
+        # Remove negative variance (because we can have very low count levels)
+        bad = np.where(exposure.variance.array < 0)
+        exposure.variance.array[bad] = np.inf
```

Comment:
See my earlier comment about the rationale of assigning infinity to a variance value already assigned as bad. It's a good idea to choose a single value, but why not another nonsensical number such as -1? Would that value result in the same failure?

Reply:
The variance must be non-negative. An infinite variance gives the pixel no weight in calculations without any additional effort, even if the user isn't respecting the mask.

Reply:
Understood.